Sources/JWTKit/RSA/RSA.swift

```diff
@@ -0,0 +1,263 @@
+        }
+
+        /// Creates an ``RSA.PublicKey`` from public certificate PEM file.
+        ///
+        /// Certificate PEM files look like:
+        ///
+        ///     -----BEGIN CERTIFICATE-----
+        ///     MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQC0cOtPjzABybjzm3fCg1aCYwnx
+        ///     ...
+        ///     aX4rbSL49Z3dAQn8vQIDAQAB
+        ///     -----END CERTIFICATE-----
+        ///
+        /// This key can only be used to verify JWTs.
+        ///
+        /// - parameters:
+        ///     - pem: Contents of PEM file.
+        public init(certificatePEM: some DataProtocol) throws {
+            let string = String(decoding: certificatePEM, as: UTF8.self)
+            try self.init(certificatePEM: string)
+        }
+
+        /// Initializes a new RSA key instance with modulus and exponent
+        ///
+        /// This initializer creates an RSA key using the modulus and exponent.
+        /// All these parameters are expected to be base64 URL encoded strings.
+        ///
+        /// - Parameters:
+        ///   - modulus: The modulus of the RSA key, represented as a base64 URL encoded string.
+        ///   - exponent: The exponent of the RSA key, represented as a base64 URL encoded string.
+        ///
+        /// - Throws:
+        ///   - ``JWTError/generic`` with the identifier `RSAKey`` if either the modulus or exponent cannot be decoded from their base64 URL encoded strings.
+        ///
+        /// - Note:
+        ///   - The provided modulus and exponent are key components for creating RSA public keys.
+        ///   - The private exponent is an additional parameter required for creating RSA private keys.
+        public init(
+            modulus: String,
+            exponent: String
+        ) throws {
+            let n = try decode(modulus)
+            let e = try decode(exponent)
+
+            var serializer = DER.Serializer()
+
+            let publicKeyDER = try RSA.calculateDER(n: n, e: e)
+            try publicKeyDER.serialize(into: &serializer)
+            let publicKey = try _RSA.Signing.PublicKey(derRepresentation: serializer.serializedBytes)
+            self.init(backing: publicKey)
+        }
+
+        package init(backing: _RSA.Signing.PublicKey) {
+            self.backing = backing
+        }
+
+        func isValidSignature<D: Digest>(_ signature: _RSA.Signing.RSASignature, for digest: D, padding: _RSA.Signing.Padding) -> Bool {
+            return self.backing.isValidSignature(signature, for: digest, padding: padding)
+        }
+    }
+}
+
+public extension RSA {
+    struct PrivateKey: RSAKey {
+        /// Exports the current private key as a PEM encoded string.
+        ///
+        /// - Throws: If the key is not a private key.
+        /// - Returns: A PEM encoded string representation of the key.
+        public var pemRepresentation: String {
+            backing.pemRepresentation
+        }
+
+        package let backing: _RSA.Signing.PrivateKey
```

Comment:
(`@inlinable` is probably not needed for most of JWTKit, too much complex math going on most of the time for it to matter, but there are some strategic places it would be helpful, we should go over that later)

Reply:
Unfortunately it's not that easy as `self.backing.publicKey` is of (SwiftCrypto's) type `_RSA.Signing.PublicKey`, this would be 
```swift
@inlinable
public var publicKey: PublicKey {
    .init(backing: self.backing.publicKey)
}
```
Not sure that's ideal (or even doable as the `init(backing:)` is internal